v4/internal/expect.go

```diff
@@ -91,21 +99,25 @@ type WantTxn struct {
 	NumErrors int
 }
 
+// WantSpan is a span or transaction expectation.
+type WantSpan struct {
+	Name       string
+	SpanID     string
+	TraceID    string
+	ParentID   string
+	Attributes map[string]interface{}
+}
+
 // Expect exposes methods that allow for testing whether the correct data was
 // captured.
 type Expect interface {
 	ExpectCustomEvents(t Validator, want []WantEvent)
 	ExpectErrors(t Validator, want []WantError)
 	ExpectErrorEvents(t Validator, want []WantEvent)
-
-	ExpectTxnEvents(t Validator, want []WantEvent)
-
 	ExpectMetrics(t Validator, want []WantMetric)
 	ExpectMetricsPresent(t Validator, want []WantMetric)
 	ExpectTxnMetrics(t Validator, want WantTxn)
-
 	ExpectTxnTraces(t Validator, want []WantTxnTrace)
 	ExpectSlowQueries(t Validator, want []WantSlowQuery)
-
-	ExpectSpanEvents(t Validator, want []WantEvent)
+	ExpectSpanEvents(t Validator, want []WantSpan)
```

Comment:
Instead of a WantEvent, this now accepts a WantSpan which has several new fields that WantEvent did not have.